### ISSUE TITLE
Sorting results by field in pivot table

### ISSUE THREAD

**Issue by MarcosAvatar** [avenirer/CodeIgniter-MY_Model#228]
Hi. I can't find a way to sort results by a field in the pivot table.

I'm using Grocery CRUD which in the n-n relationships uses a table with an order field, like {table_one_id, table_two_id, priority} in which "priority" is used for the order of the related rows.

I know there is "order_inside", which sorts the related results by a field in the related table, but the field I need to use for sorting is in the pivot table.

**Comment by salain** [comment]
A pivot table is only use to "link" the table and does not / should not have any other information. If you want to keep you priority field in that table I think you will have to right the query the old fashion way.

**Comment by MarcosAvatar** [comment]
**salain**, thanks for your collaboration.

Ok. I understand. The problem was that the admin panel was already finished using that Grocery CRUD feature, and I'm on the limit with this project.

I fiddled with the "join:true" property (issue: https://github.com/avenirer/CodeIgniter-MY_Model/issues/207), but I was unable to make use of it in this case.

If it helps someone, the solution I found (not a pretty one, but a fast one) was using the "order_inside" property such as "order_inside: pivot_table.order_column ASC", as order_inside is an "ORDER BY" clause used within the query which makes the join with the pivot and the related table.

The only problem with this was that the code _always_ prepends the related table name to the "order_inside" parameter, so I added a few lines to check if the property has a dot within, if it has, it doesn't prepend the table name assuming you're referencing the table yourself.

Sorry I had to do this, but Grocery CRUD is widely used and solves in a simple way the sorting order issue. I think this would be a nice feature that allows you to use both and this fix doesn't alter its main functionality.

Thanks!
